src/api/app/models/user.rb

```diff
@@ -842,11 +842,11 @@ def to_param
     to_s
   end
 
-  def nr_of_requests_that_need_work
-    Rails.cache.fetch("requests_for_#{login}", expires_in: 2.minutes) do
-      BsRequest.collection(user: login, states: %w(declined), roles: %w(creator)).count +
-      BsRequest.collection(user: login, states: %w(new), roles: %w(maintainer)).count +
-      BsRequest.collection(user: login, roles: %w(reviewer), reviewstates: %w(new), states: %w(review)).count
+  def tasks
+    Rails.cache.fetch("requests_for_#{cache_key}") do
+      declined_requests.count +
+      incoming_requests.count +
+      involved_reviews.count
```

Comment:
hm, I think we are in less important corner cases here. But according to the method 
name nr_of_requests_that_need_work the technical correct implementation would be

 (declined_requests + incoming_requests + involved_reviews.map(:request)).uniq.count

this would also obsolete your question ;)

however, this method use for showing the number of tasks for the user, so the current implementation is correct and also the request where the user is the creator _and_ responsible reviewer should be counted. Since the use shall act on this review  in the request.

However, again, these are corner cases and just for counting. It is more important that the user sees all tasks. Is the review task listed in the users task list when he is creator and reviewer?